source/common/dynamo/dynamo_request_parser.cc

```diff
@@ -125,4 +125,28 @@ bool RequestParser::isBatchOperation(const std::string& operation) {
          BATCH_OPERATIONS.end();
 }
 
+std::vector<RequestParser::PartitionDescriptor>
+RequestParser::parsePartitions(const std::string& data) {
+  Json::StringLoader json(data);
+  std::vector<RequestParser::PartitionDescriptor> partition_descriptors;
+
+  if (json.hasObject("ConsumedCapacity")) {
+    Json::Object consumed_capacity = json.getObject("ConsumedCapacity");
+    if (consumed_capacity.hasObject("Partitions")) {
+      Json::Object partitions = consumed_capacity.getObject("Partitions");
+
+      partitions.iterate(
+          [&partition_descriptors, &partitions](const std::string& key, const Json::Object&) {
+            // Capacity is a double and it is rounded up to the nearest integer.
```

Comment:
it's not intuitive to me what is going on here. Since this is not yet in the public docs, can you have a slightly larger comment on what the API returns here and how we are processing it